JS/Lesson34/goksel/recipe-app/src/modules/NewRecipe/NewRecipeForm/index.jsx

```diff
@@ -0,0 +1,151 @@
+import { onSubmit } from "../NewRecipeInformations";
+import "./styles.css";
+
+export const NewRecipe = () => {
+
+    const {
+        register,
+        handleSubmit,
+        formState: { errors },
+    } = useForm();
+
+    return (
+        <form
+            className="recipe-form"
+            noValidate
+            onSubmit={handleSubmit(onSubmit)}>
+            <label htmlFor="strMeal">Recipe Name</label>
+            <input
+                id="strMeal"
+                type="text"
+                {...register("strMeal", { required: "Recipe name is required", minLength: { value: 5, message: "Recipe Name should be at least 5 characters long" } })}
+            />
+            {errors.strMeal && (
+                <span className="error-message">{errors.strMeal.message}</span>
+            )}
+
+            <label htmlFor="strDrinkAlternate">Drink Alternate</label>
+            <input
+                id="strDrinkAlternate"
+                type="text"
+                {...register("strDrinkAlternate")}
+            />
+
+            <label htmlFor="strCategory">Category</label>
+            <select id="strCategory" {...register("strCategory", { required: true })}>
+                <option disabled selected>Select Your Category</option>
+                {MealCategory.map((category) => (
+                        <option key={category.id}>{category.title}</option>
+                    ))}
+            </select>
+
+            <label htmlFor="strArea">Meal Area</label>
+            <select id="strArea" {...register("strArea", { required: true })}>
+                <option disabled selected>Select Your Area</option>
+                {MealArea.map((area) => (
+                    <option key={area.id}>{area.title}</option>
+                ))}
+            </select>
+
+            <label htmlFor="strInstructions">Instractions</label>
+            <textarea
+                id="strInstructions"
+                {...register("strInstructions", {
+                    required: "Instractions are required",
+                    minLength: {
+                        value: 50,
+                        message: "Please enter Instractions at least 50 characters long"
+                    }
+                })}
+            />
+            {errors.strInstructions && (
+                <span className="error-message">{errors.strInstructions.message}</span>
+            )}
+
+            <label htmlFor="strMealThumb">Upload Your Image</label>
+            <input
+                id="strMealThumb"
+                type="file"
+                {...register("strMealThumb")}
+            />
+
+            <label htmlFor="strTags">Main Product</label>
+            <textarea
+                id="strTags"
+                {...register("strTags", {
+                    required: "Tags are required",
+                    minLength: {
+                        value: 5,
+                        message: "Please enter at least 1 tag"
```

Comment:
This is a bit confusing since I can enter one tag that is less than 5 characters but will still see error saying "Please enter at least 1 tag". Instead, say something like 'Tags should be at least X characters long'. I'd recommend to lower the limit to 3 characters.